crates/polars-sql/src/context.rs

```diff
@@ -107,6 +115,23 @@ impl SQLContext {
         self.cte_map.borrow_mut().clear();
         res
     }
+
+    /// add a function registry to the SQLContext
+    /// the registry provides the ability to add custom functions to the SQLContext
+    pub fn with_function_registry(mut self, function_registry: Arc<dyn FunctionRegistry>) -> Self {
+        self.function_registry = function_registry;
+        self
```

Comment:
If I was using this, I'd assume that the registries would combine, not overwrite. When you do `with_context`, that's how it works.

Reply:
a lot of the other structs such as the readers/writers use this pattern & is a pretty common alternative to the builder pattern. 
- [csv reader](https://github.com/universalmind303/polars/blob/410f869de8356df381d8ee3c7cbc38186dd76e04/crates/polars-io/src/csv/read.rs#L139)
- [csv writer](https://github.com/universalmind303/polars/blob/410f869de8356df381d8ee3c7cbc38186dd76e04/crates/polars-io/src/csv/write.rs#L60)
- [cloud options](https://github.com/universalmind303/polars/blob/410f869de8356df381d8ee3c7cbc38186dd76e04/crates/polars-core/src/cloud.rs#L94)